Organize sections (#3481)
qgis/QGIS-Documentation: source/docs/pyqgis_developer_cookbook/vector.rst

.. index:: Vector layers; Creating

> Creating Vector Layers

Reviewer: ```suggestion
Creating and Exporting Vector Layers
```

> ======================

Reviewer: ```suggestion
===================================
```

> You can create vector layer datasets using the :class:`QgsVectorFileWriter` class.

Reviewer: ```suggestion
You can create vector layers using the :class:`QgsVectorLayer` class and export as vector datasets using the :class:`QgsVectorFileWriter` class.
```

> It supports all the vector formats that OGR supports (GeoPackage, Shapefile,

Reviewer: ```suggestion
The :class:`QgsVectorLayer` class can be used to create temporary, memory-based layers (`memory`) and
connect to OGR datasets (`ogr`), databases (`postgres`, `spatialite`, `mysql`) and more (`wfs`, `gpx`, `delimitedtext`, ...).
The :class:`QgsVectorFileWriter` class supports all OGR supported vector formats (GeoPackage,
Shapefile,
```

> supports any other kind of vector file that OGR supports (GeoPackage, Shapefile, GeoJSON,

Reviewer: ```suggestion
supports any other kind of vector format that OGR supports (GeoPackage, Shapefile, GeoJSON,
```

> KML and others).

Author: I suspect this introductory paragraph might need a revisit but can't do that.

> There are different possibilities how to export a vector layer.

Reviewer: ```suggestion
There are several ways to export a vector layer.
```

  GeoJSON, KML and others).
* the :class:`QgsVectorLayer <qgis.core.QgsVectorLayer>` class: instantiates
  a data provider that interprets the supplied path (url) of the data source
  to connect to and access the data. It can be used to create temporary,
  memory-based layers (``memory``) and connect to OGR datasets (``ogr``),
  databases (``postgres``, ``spatialite``, ``mysql``, ``mssql``) and
  more (``wfs``, ``gpx``, ``delimitedtext``...).

From an instance of :class:`QgsVectorFileWriter`
------------------------------------------------

.. code-block:: python

  error = QgsVectorFileWriter.writeAsVectorFormat(layer, "my_data", "UTF-8")
  if error[0] == QgsVectorFileWriter.NoError:
      print("success!")

  error = QgsVectorFileWriter.writeAsVectorFormat(layer, "my_json", "UTF-8",  driverName="GeoJSON")
  if error[0] == QgsVectorFileWriter.NoError:
      print("success again!")

The third parameter specifies output text encoding. Only some drivers need this
for correct operation (Shapefile is one of those), but if you
are not using international characters you do not have to care much about
the encoding.

> The fourth parameter that we left as ``None`` may specify destination CRS ---

Reviewer: ```suggestion
The fourth parameter that we left as ``None`` may specify the destination CRS ---
```
